### ISSUE TITLE
InstrumentationSubscriberError

### ISSUE THREAD

**Issue by kurenn** [airbrake/airbrake#1205]
While trying to upgrade to Rails 7, I'm encountering and error with my test suite. My current setup is:

- Rails `7.0.0`
- Capybara `3.35`
- Cuprite `~> 0.13`

And when I run my system tests I get this output:

```
ActiveSupport::Notifications::InstrumentationSubscriberError:
       Exception(s) occurred within instrumentation subscribers: NoMethodError, NoMethodError
     
     [Screenshot Image]: tmp/capybara/screenshots/failures_r_spec_example_groups_destroy_candidate_when_the_user_is_logged_in_destroys_the_record_successfully_864.png

     
     # /usr/local/bundle/gems/warden-1.2.9/lib/warden/manager.rb:36:in `block in call'
     # /usr/local/bundle/gems/warden-1.2.9/lib/warden/manager.rb:34:in `catch'
     # /usr/local/bundle/gems/warden-1.2.9/lib/warden/manager.rb:34:in `call'
     # /usr/local/bundle/gems/rack-2.2.3/lib/rack/tempfile_reaper.rb:15:in `call'
     # /usr/local/bundle/gems/rack-2.2.3/lib/rack/etag.rb:27:in `call'
     # /usr/local/bundle/gems/rack-2.2.3/lib/rack/conditional_get.rb:27:in `call'
     # /usr/local/bundle/gems/rack-2.2.3/lib/rack/head.rb:12:in `call'
     # /usr/local/bundle/gems/rack-2.2.3/lib/rack/session/abstract/id.rb:266:in `context'
     # /usr/local/bundle/gems/rack-2.2.3/lib/rack/session/abstract/id.rb:260:in `call'
     # /usr/local/bundle/gems/airbrake-12.0.0/lib/airbrake/rack/middleware.rb:34:in `call!'
     # /usr/local/bundle/gems/airbrake-12.0.0/lib/airbrake/rack/middleware.rb:23:in `call'
     # /usr/local/bundle/gems/railties-7.0.0/lib/rails/rack/logger.rb:36:in `call_app'
     # /usr/local/bundle/gems/railties-7.0.0/lib/rails/rack/logger.rb:25:in `block in call'
     # /usr/local/bundle/gems/railties-7.0.0/lib/rails/rack/logger.rb:25:in `call'
     # /usr/local/bundle/gems/rack-2.2.3/lib/rack/method_override.rb:24:in `call'
     # /usr/local/bundle/gems/rack-2.2.3/lib/rack/runtime.rb:22:in `call'
     # /usr/local/bundle/gems/rack-2.2.3/lib/rack/sendfile.rb:110:in `call'
     # /usr/local/bundle/gems/railties-7.0.0/lib/rails/engine.rb:530:in `call'
     # /usr/local/bundle/gems/rack-2.2.3/lib/rack/urlmap.rb:74:in `block in call'
     # /usr/local/bundle/gems/rack-2.2.3/lib/rack/urlmap.rb:58:in `each'
     # /usr/local/bundle/gems/rack-2.2.3/lib/rack/urlmap.rb:58:in `call'
     # /usr/local/bundle/gems/rack-2.2.3/lib/rack/builder.rb:244:in `call'
     # /usr/local/bundle/gems/capybara-3.35.3/lib/capybara/server/middleware.rb:60:in `call'
     # /usr/local/bundle/gems/puma-5.5.1/lib/puma/configuration.rb:249:in `call'
     # /usr/local/bundle/gems/puma-5.5.1/lib/puma/request.rb:77:in `block in handle_request'
     # /usr/local/bundle/gems/puma-5.5.1/lib/puma/thread_pool.rb:340:in `with_force_shutdown'
     # /usr/local/bundle/gems/puma-5.5.1/lib/puma/request.rb:76:in `handle_request'
     # /usr/local/bundle/gems/puma-5.5.1/lib/puma/server.rb:447:in `process_client'
     # /usr/local/bundle/gems/puma-5.5.1/lib/puma/thread_pool.rb:147:in `block in spawn_thread'
     # ------------------
     # --- Caused by: ---
     # NoMethodError:
     #   undefined method `getutc' for 1640053493229.386:Float
     #   /usr/local/bundle/gems/airbrake-ruby-6.0.1/lib/airbrake-ruby/time_truncate.rb:12:in `utc_truncate_minutes'
```

Why I'm looking at, is:

`#   undefined method `getutc' for 1640053493229.386:Float`

Due to the rails change(or at least I can infer that), now instead of a `Time` object is now sending a `Float`.

If I disable Airbrake, everything works as expected

Thanks for the support, and let me know if I can be of more help

**Comment by hobbypunk90** [comment]
I have the same error on a plain rails 7 application.
The moment i add airbrake to the gemfile, it's not working anymore:
`
ActiveSupport::Notifications::InstrumentationSubscriberError (Exception(s) occurred within instrumentation subscribers: NoMethodError, NoMethodError):
  
activesupport (7.0.0) lib/active_support/notifications/fanout.rb:101:in `iterate_guarding_exceptions'
activesupport (7.0.0) lib/active_support/notifications/fanout.rb:76:in `finish'
activesupport (7.0.0) lib/active_support/notifications/instrumenter.rb:49:in `finish_with_state'
activesupport (7.0.0) lib/active_support/notifications/instrumenter.rb:30:in `instrument'
activesupport (7.0.0) lib/active_support/notifications.rb:206:in `instrument'
actionpack (7.0.0) lib/action_controller/metal/instrumentation.rb:66:in `process_action'
actionpack (7.0.0) lib/action_controller/metal/params_wrapper.rb:259:in `process_action'
activerecord (7.0.0) lib/active_record/railties/controller_runtime.rb:27:in `process_action'
actionpack (7.0.0) lib/abstract_controller/base.rb:151:in `process'
actionview (7.0.0) lib/action_view/rendering.rb:39:in `process'
actionpack (7.0.0) lib/action_controller/metal.rb:188:in `dispatch'
actionpack (7.0.0) lib/action_controller/metal.rb:251:in `dispatch'
actionpack (7.0.0) lib/action_dispatch/routing/route_set.rb:49:in `dispatch'
actionpack (7.0.0) lib/action_dispatch/routing/route_set.rb:32:in `serve'
actionpack (7.0.0) lib/action_dispatch/journey/router.rb:50:in `block in serve'
actionpack (7.0.0) lib/action_dispatch/journey/router.rb:32:in `each'
actionpack (7.0.0) lib/action_dispatch/journey/router.rb:32:in `serve'
actionpack (7.0.0) lib/action_dispatch/routing/route_set.rb:850:in `call'
omniauth (2.0.4) lib/omniauth/strategy.rb:202:in `call!'
omniauth (2.0.4) lib/omniauth/strategy.rb:169:in `call'
warden (1.2.9) lib/warden/manager.rb:36:in `block in call'
warden (1.2.9) lib/warden/manager.rb:34:in `catch'
warden (1.2.9) lib/warden/manager.rb:34:in `call'
rack (2.2.3) lib/rack/tempfile_reaper.rb:15:in `call'
rack (2.2.3) lib/rack/etag.rb:27:in `call'
rack (2.2.3) lib/rack/conditional_get.rb:27:in `call'
rack (2.2.3) lib/rack/head.rb:12:in `call'
actionpack (7.0.0) lib/action_dispatch/http/permissions_policy.rb:22:in `call'
actionpack (7.0.0) lib/action_dispatch/http/content_security_policy.rb:18:in `call'
rack (2.2.3) lib/rack/session/abstract/id.rb:266:in `context'
rack (2.2.3) lib/rack/session/abstract/id.rb:260:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/cookies.rb:693:in `call'
activerecord (7.0.0) lib/active_record/migration.rb:603:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
activesupport (7.0.0) lib/active_support/callbacks.rb:99:in `run_callbacks'
actionpack (7.0.0) lib/action_dispatch/middleware/callbacks.rb:26:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/actionable_exceptions.rb:17:in `call'
airbrake (12.0.0) lib/airbrake/rack/middleware.rb:34:in `call!'
airbrake (12.0.0) lib/airbrake/rack/middleware.rb:23:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/debug_exceptions.rb:28:in `call'
web-console (4.2.0) lib/web_console/middleware.rb:132:in `call_app'
web-console (4.2.0) lib/web_console/middleware.rb:28:in `block in call'
web-console (4.2.0) lib/web_console/middleware.rb:17:in `catch'
web-console (4.2.0) lib/web_console/middleware.rb:17:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/show_exceptions.rb:26:in `call'
railties (7.0.0) lib/rails/rack/logger.rb:36:in `call_app'
railties (7.0.0) lib/rails/rack/logger.rb:25:in `block in call'
activesupport (7.0.0) lib/active_support/tagged_logging.rb:99:in `block in tagged'
activesupport (7.0.0) lib/active_support/tagged_logging.rb:37:in `tagged'
activesupport (7.0.0) lib/active_support/tagged_logging.rb:99:in `tagged'
railties (7.0.0) lib/rails/rack/logger.rb:25:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/remote_ip.rb:93:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/request_id.rb:26:in `call'
rack (2.2.3) lib/rack/method_override.rb:24:in `call'
rack (2.2.3) lib/rack/runtime.rb:22:in `call'
activesupport (7.0.0) lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/server_timing.rb:20:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/static.rb:23:in `call'
rack (2.2.3) lib/rack/sendfile.rb:110:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/host_authorization.rb:137:in `call'
railties (7.0.0) lib/rails/engine.rb:530:in `call'
puma (5.5.2) lib/puma/configuration.rb:249:in `call'
puma (5.5.2) lib/puma/request.rb:77:in `block in handle_request'
puma (5.5.2) lib/puma/thread_pool.rb:340:in `with_force_shutdown'
puma (5.5.2) lib/puma/request.rb:76:in `handle_request'
puma (5.5.2) lib/puma/server.rb:447:in `process_client'
puma (5.5.2) lib/puma/thread_pool.rb:147:in `block in spawn_thread'
`

**Comment by jlestavel** [comment]
Same error here. Here is a minimal script to reproduce :

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  gem "rails", "~> 7.0"
  gem "airbrake"
end

require "rack/test"
require "action_controller/railtie"

class TestApp < Rails::Application
  config.root = __dir__
  config.hosts << "example.org"
  config.session_store :cookie_store, key: "cookie_store_key"
  secrets.secret_key_base = "secret_key_base"
  config.action_controller.default_protect_from_forgery = true

  config.logger = Logger.new($stdout)
  Rails.logger  = config.logger

  routes.draw do
    get "/" => "test#index"
  end
end

class TestController < ActionController::Base
  def index
    render plain: "Home"
  end
end

TestApp.initialize!

require "minitest/autorun"

class BugTest < Minitest::Test
  include Rack::Test::Methods

  def test_returns_success
    get "/"
    assert last_response.ok?
  end

  private
    def app
      Rails.application
    end
end
```
And here is the output of the test :

```
# Running:

I, [2021-12-21T14:12:22.325658 #31000]  INFO -- : Started GET "/" for 127.0.0.1 at 2021-12-21 14:12:22 +0100
I, [2021-12-21T14:12:22.347495 #31000]  INFO -- : Processing by Rails::WelcomeController#index as HTML
D, [2021-12-21T14:12:22.359047 #31000] DEBUG -- :   Rendering /Users/jlestavel/.rbenv/versions/3.0.1/lib/ruby/gems/3.0.0/gems/railties-7.0.0/lib/rails/templates/rails/welcome/index.html.erb
I, [2021-12-21T14:12:22.361406 #31000]  INFO -- :   Rendered /Users/jlestavel/.rbenv/versions/3.0.1/lib/ruby/gems/3.0.0/gems/railties-7.0.0/lib/rails/templates/rails/welcome/index.html.erb (Duration: 2.2ms | Allocations: 504)
I, [2021-12-21T14:12:22.361805 #31000]  INFO -- : Completed 200 OK in 14ms (Views: 6.6ms | Allocations: 2776)


F, [2021-12-21T14:12:22.364260 #31000] FATAL -- :   
NoMethodError (undefined method `getutc' for 1640092342347.901:Float):
  
airbrake-ruby (6.0.1) lib/airbrake-ruby/time_truncate.rb:12:in `utc_truncate_minutes'
airbrake-ruby (6.0.1) lib/airbrake-ruby/request.rb:23:in `initialize'
airbrake-ruby (6.0.1) lib/airbrake-ruby.rb:392:in `new'
airbrake-ruby (6.0.1) lib/airbrake-ruby.rb:392:in `notify_request'
airbrake (12.0.0) lib/airbrake/rails/action_controller_notify_subscriber.rb:21:in `block in call'
airbrake (12.0.0) lib/airbrake/rails/action_controller_notify_subscriber.rb:20:in `each'
airbrake (12.0.0) lib/airbrake/rails/action_controller_notify_subscriber.rb:20:in `call'
activesupport (7.0.0) lib/active_support/notifications/fanout.rb:236:in `finish'
activesupport (7.0.0) lib/active_support/notifications/fanout.rb:76:in `block in finish'
activesupport (7.0.0) lib/active_support/notifications/fanout.rb:91:in `block in iterate_guarding_exceptions'
activesupport (7.0.0) lib/active_support/notifications/fanout.rb:90:in `each'
activesupport (7.0.0) lib/active_support/notifications/fanout.rb:90:in `iterate_guarding_exceptions'
activesupport (7.0.0) lib/active_support/notifications/fanout.rb:76:in `finish'
activesupport (7.0.0) lib/active_support/notifications/instrumenter.rb:49:in `finish_with_state'
activesupport (7.0.0) lib/active_support/notifications/instrumenter.rb:30:in `instrument'
activesupport (7.0.0) lib/active_support/notifications.rb:206:in `instrument'
actionpack (7.0.0) lib/action_controller/metal/instrumentation.rb:66:in `process_action'
actionpack (7.0.0) lib/action_controller/metal/params_wrapper.rb:259:in `process_action'
actionpack (7.0.0) lib/abstract_controller/base.rb:151:in `process'
actionview (7.0.0) lib/action_view/rendering.rb:39:in `process'
actionpack (7.0.0) lib/action_controller/metal.rb:188:in `dispatch'
actionpack (7.0.0) lib/action_controller/metal.rb:251:in `dispatch'
actionpack (7.0.0) lib/action_dispatch/routing/route_set.rb:49:in `dispatch'
actionpack (7.0.0) lib/action_dispatch/routing/route_set.rb:32:in `serve'
actionpack (7.0.0) lib/action_dispatch/journey/router.rb:50:in `block in serve'
actionpack (7.0.0) lib/action_dispatch/journey/router.rb:32:in `each'
actionpack (7.0.0) lib/action_dispatch/journey/router.rb:32:in `serve'
actionpack (7.0.0) lib/action_dispatch/routing/route_set.rb:850:in `call'
rack (2.2.3) lib/rack/tempfile_reaper.rb:15:in `call'
rack (2.2.3) lib/rack/etag.rb:27:in `call'
rack (2.2.3) lib/rack/conditional_get.rb:27:in `call'
rack (2.2.3) lib/rack/head.rb:12:in `call'
actionpack (7.0.0) lib/action_dispatch/http/permissions_policy.rb:22:in `call'
actionpack (7.0.0) lib/action_dispatch/http/content_security_policy.rb:18:in `call'
rack (2.2.3) lib/rack/session/abstract/id.rb:266:in `context'
rack (2.2.3) lib/rack/session/abstract/id.rb:260:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/cookies.rb:693:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
activesupport (7.0.0) lib/active_support/callbacks.rb:99:in `run_callbacks'
actionpack (7.0.0) lib/action_dispatch/middleware/callbacks.rb:26:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/executor.rb:14:in `call'
airbrake (12.0.0) lib/airbrake/rack/middleware.rb:34:in `call!'
airbrake (12.0.0) lib/airbrake/rack/middleware.rb:23:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/debug_exceptions.rb:28:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/show_exceptions.rb:26:in `call'
railties (7.0.0) lib/rails/rack/logger.rb:36:in `call_app'
railties (7.0.0) lib/rails/rack/logger.rb:27:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/remote_ip.rb:93:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/request_id.rb:26:in `call'
rack (2.2.3) lib/rack/method_override.rb:24:in `call'
rack (2.2.3) lib/rack/runtime.rb:22:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/static.rb:23:in `call'
rack (2.2.3) lib/rack/sendfile.rb:110:in `call'
actionpack (7.0.0) lib/action_dispatch/middleware/host_authorization.rb:137:in `call'
railties (7.0.0) lib/rails/engine.rb:530:in `call'
rack-test (1.1.0) lib/rack/mock_session.rb:29:in `request'
rack-test (1.1.0) lib/rack/test.rb:266:in `process_request'
rack-test (1.1.0) lib/rack/test.rb:129:in `custom_request'
rack-test (1.1.0) lib/rack/test.rb:58:in `get'
/Users/jlestavel/.rbenv/versions/3.0.1/lib/ruby/3.0.0/forwardable.rb:238:in `get'
airbrake_rails7.rb:44:in `test_returns_success'
minitest (5.15.0) lib/minitest/test.rb:98:in `block (3 levels) in run'
minitest (5.15.0) lib/minitest/test.rb:195:in `capture_exceptions'
minitest (5.15.0) lib/minitest/test.rb:95:in `block (2 levels) in run'
minitest (5.15.0) lib/minitest.rb:281:in `time_it'
minitest (5.15.0) lib/minitest/test.rb:94:in `block in run'
minitest (5.15.0) lib/minitest.rb:376:in `on_signal'
minitest (5.15.0) lib/minitest/test.rb:221:in `with_info_handler'
minitest (5.15.0) lib/minitest/test.rb:93:in `run'
minitest (5.15.0) lib/minitest.rb:1042:in `run_one_method'
minitest (5.15.0) lib/minitest.rb:350:in `run_one_method'
minitest (5.15.0) lib/minitest.rb:337:in `block (2 levels) in run'
minitest (5.15.0) lib/minitest.rb:336:in `each'
minitest (5.15.0) lib/minitest.rb:336:in `block in run'
minitest (5.15.0) lib/minitest.rb:376:in `on_signal'
minitest (5.15.0) lib/minitest.rb:363:in `with_info_handler'
minitest (5.15.0) lib/minitest.rb:335:in `run'
minitest (5.15.0) lib/minitest.rb:169:in `block in __run'
minitest (5.15.0) lib/minitest.rb:169:in `map'
minitest (5.15.0) lib/minitest.rb:169:in `__run'
minitest (5.15.0) lib/minitest.rb:146:in `run'
minitest (5.15.0) lib/minitest.rb:73:in `block in autorun'
F

Failure:
BugTest#test_returns_success [airbrake_rails7.rb:45]:
Expected false to be truthy.


rails test airbrake_rails7.rb:43



Finished in 0.061137s, 16.3567 runs/s, 16.3567 assertions/s.
1 runs, 1 assertions, 1 failures, 0 errors, 0 skips
```
If you change the Rails version to `"~> 6.0"` it works well. If you remove the `airbrake` gem, it works too.

**Comment by tnir** [comment]
Had the same error in my app with Rails 7.0.0 and airbrake 12.0.0 / airbrake-ruby 6.0.1. Just prepared a patch in airbrake/airbrake-ruby#673.

**Comment by kurenn** [comment]
Thanks for the patch! I'll test it and come back to you

**Comment by kurenn** [comment]
> Thanks for the patch! I'll test it and come back to you

I can confirm that it is working for me

**Comment by kyrylo** [comment]
Thanks everyone for chiming in. I just released airbrake-ruby v6.0.2, which should fix this issue. Please `bundle update airbrake-ruby` and you should be good to go.
